src/gpl/src/solver.h

```diff
@@ -0,0 +1,85 @@
+#ifndef __SOLVER__
+#define __SOLVER__
+
+#include <Eigen/IterativeLinearSolvers>
+#include <Eigen/SparseCore>
+#include <memory>
+
+#ifdef ENABLE_GPU
+#include "gpuSolver.h"
+#endif
+#include "graphics.h"
+#include "odb/db.h"
+#include "placerBase.h"
+#include "plot.h"
+#include "utl/Logger.h"
+
+namespace utl {
+class Logger;
+}
+
+namespace gpl {
+
+using Eigen::BiCGSTAB;
+using Eigen::IdentityPreconditioner;
+using utl::GPL;
+
+using namespace std;
+typedef Eigen::SparseMatrix<float, Eigen::RowMajor> SMatrix;
+
+#ifdef ENABLE_GPU
+ void cudaSparseSolve(int iter,
+                     SMatrix& placeInstForceMatrixX,
+                     Eigen::VectorXf& fixedInstForceVecX,
+                     Eigen::VectorXf& instLocVecX,
+                     SMatrix& placeInstForceMatrixY,
+                     Eigen::VectorXf& fixedInstForceVecY,
+                     Eigen::VectorXf& instLocVecY,
+                     float errorX,
+                     float errorY,
+                     utl::Logger* logger,
+                     int hpwl)
+{
+  class GpuSolver SP1(placeInstForceMatrixX, fixedInstForceVecX, logger);
```

Comment:
Why is the keyword class here?  same for SP2

Reply:
It is a bad typo... I didn't see any errors here. So didn't notice there exists a weird "class". Deleted it now.